nixos/modules/services/web-apps/immich.md

```diff
@@ -0,0 +1,276 @@
+```
+
+Next you should create a backup of your existing database. While this
+guide is not going to modify the original database it might still be
+advisable in case something goes wrong.
+
+Example, assuming your Immich database is running in a container
+named `immich_postgres`:
+
+```console
+# docker exec immich_postgres pg_dump --username immich immich -f /tmp/immich.sql > /var/backup/immich.sql
+```
+
+::: {.warning}
+Do not add `-t`/`--tty` to the `docker` command when using it in a
+pipe. Using this option will combine STDOUT and STDERR as well as
+potentially print control characters. This is undesirable when dumping
+a database, as output other than the actual SQL commands might get
+written to the database dump.
+:::
+
+It is advised to store these backups in a safe location, at least
+until you have confirmed that your new installation works.
+
+### Configure Immich NixOS module {#module-immich-migrating-from-docker-compose-configure}
+
+You can now enable Immich in your NixOS configuration as described in
+[Configuring](#module-immich-configuring).
+
+After building and switching to the new configuration, you should
+stop the newly started `immich-server.service` and
+`immich-machine-learning.service` units.
+
+```console
+# systemctl stop immich-server.service immich-machine-learning.service
+```
+
+### Create database dump {#module-immich-migrating-from-docker-compose-dump-database}
+
+You might have done this already during your
+[backup step](#module-immich-migrating-from-docker-compose-backup).
+You can choose to reuse that database dump by making a copy or create
+another dump as described in that section.
+
+The remainder of this guide assumes that the copy of the database dump
+is located at `./immich.sql`.
+
+::: {.note}
+While `pg_dump` offers a few output formats, this guide relies on the
+`plain` (default) format, as it will be modified in later steps.
+:::
+
+### Fix up database dump {#module-immich-migrating-from-docker-compose-fixup-dump}
+
+The database dump generated by `pg_dump`, can most likely not be
+imported as is. There are two changes we need to make, to allow it to
+be imported correctly.
+
+<!-- TODO: Is the following actually a Postgres bug? -->
+First, we need to restore value of the `search_path` variable of the
+dump, as PostgreSQL will otherwise be unable to import data that
+makes use of its `earthdistance` extension.
+See [this StackOverflow comment](https://stackoverflow.com/questions/49702964/postgres-import-to-rds-could-not-execute-query-error-type-earth-does-not-ex#comment86441215_49702964)
+and [Immich "Backup and Restore" documentation](https://immich.app/docs/administration/backup-and-restore/)
+for more information.
+
+To work around this issue, we can adjust the `search_path` variable
+in the database dump. This variable is set within the first 20
+lines of your database dump, so you can just open up your database
+dump in your preferred text editor and apply the following changes:
```

Comment:
Opening a DB dump in a text editor has issues. It'd be better to just put another `sed` command here.

Reply:
Done